gsuite_reports_rules/gsuite_drive_visibility_change.py

```diff
@@ -1,42 +1,123 @@
         return False
 
+    #########
+    # for target_domain
+    change_document_visibility = False
+
+    for details in event.get("events", [{}]):
+        if (
+            details.get("type") == "acl_change"
+            and details.get("name") == "change_document_visibility"
+            and param_lookup(details.get("parameters", {}), "new_value") != ["private"]
+            and param_lookup(details.get("parameters", {}), "target_domain") not in EXCLUDED_DOMAINS
+            and param_lookup(details.get("parameters", {}), "visibility") in VISIBILITY
+        ):
+            global TARGET_DOMAIN
+            TARGET_DOMAIN = param_lookup(details.get("parameters", {}), "target_domain")
+            global NEW_VISIBILITY
+            NEW_VISIBILITY = param_lookup(details.get("parameters", {}), "visibility")
+            global DOC_TITLE
+            DOC_TITLE = param_lookup(details.get("parameters", {}), "doc_title")
+
+            change_document_visibility = True
+            break
+
+    # "change_document_access_scope" events are always paired with "change_document_visibility" events
+    # the "target_domain" and "visibility" attributes are the same
+    # the set of a "change_document_access_scope" and a "change_document_visibility" is paired with a
+    # second set of these events
+    # the first pair of events represents the "old" domain and visibility, and the second pair
+    # represents the "new" domain and visibility
+    if change_document_visibility:
+        for details in event.get("events", [{}]):
+            if (
+                details.get("type") == "acl_change"
+                and details.get("name") == "change_document_access_scope"
+                and param_lookup(details.get("parameters", {}), "new_value") != ["none"]
+            ):
+                global ACCESS_SCOPE
+                ACCESS_SCOPE = param_lookup(details.get("parameters", {}), "new_value")
+        return True
+
+    # TODO: confirm that change_user_access events are mutually exclusive with the above events
+    # split this out
+    #########
+    # for target_user
+    change_user_access = False
+
     for details in event.get("events", [{}]):
         if (
             details.get("type") == "acl_change"
-            and param_lookup(details.get("parameters", {}), "visibility_change") == "external"
+            and details.get("name") == "change_user_access"
+            and param_lookup(details.get("parameters", {}), "new_value") != ["none"]
         ):
-            return True
+            global TARGET_USER_EMAIL
+            TARGET_USER_EMAIL = param_lookup(details.get("parameters", {}), "target_user")
+            DOC_TITLE = param_lookup(details.get("parameters", {}), "doc_title")
+            ACCESS_SCOPE = param_lookup(details.get("parameters", {}), "new_value")
+
+            change_user_access = True
+            break
+
+    if change_user_access:
+        return True
 
     return False
 
 
 def dedup(event):
```

Comment:
per my prior suggestion

```suggestion
def dedup(_):
```